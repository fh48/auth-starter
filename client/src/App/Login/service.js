import {VALIDATION_MESSAGE} from '../../modules/constants';
import * as yup from 'yup';

export const getValidationSchema = () =>
  yup.object().shape({
    email: yup
      .string()
      .email(VALIDATION_MESSAGE.EMAIL.VALID)
      .required(VALIDATION_MESSAGE.EMAIL.REQUIRED),
    password: yup.string().required(VALIDATION_MESSAGE.PASSWORD.REQUIRED)
  });
