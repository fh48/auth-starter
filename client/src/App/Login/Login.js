import React, {Component} from 'react';
import {Formik} from 'formik';

import {login} from '../../modules/api/auth.api';

import * as service from './service';

export default class Login extends Component {
  constructor(props) {
    super(props);
    this.initialValues = {
      email: '',
      password: ''
    };
  }

  handleSubmit = async (values, {setSubmitting, setErrors}) => {
    try {
      // await login(values.email, values.password);
      console.log('User has been sucessfully logged in !', values);
      setSubmitting(false);
    } catch (error) {
      console.error(error);
      // from docs: setErrors: (fields: { [field: string]: string }) => void
    }
  };

  renderForm = props => {
    // Formik internal API, problematic with UI library components
    const {isSubmitting, errors, handleChange, handleSubmit} = props;

    return (
      <form className="form">
        <label className="form-field" htmlFor="email">
          <span>E-mail:</span>
          <input name="email" type="email" onChange={handleChange} />
        </label>

        <div className="form-field-error">{errors.email}</div>
        <label className="form-field" htmlFor="password">
          <span>Password:</span>
          <input name="password" type="password" onChange={handleChange} />
        </label>
        <div className="form-field-error">{errors.password}</div>

        <button type="submit" onClick={handleSubmit} variant="contained">
          {isSubmitting ? 'Loading' : 'Login'}
        </button>
      </form>
    );
  };

  render() {
    return (
      <Formik
        initialValues={this.state}
        validationSchema={service.getValidationSchema}
        onSubmit={this.handleSubmit}
        render={this.renderForm}
      />
    );
  }
}
