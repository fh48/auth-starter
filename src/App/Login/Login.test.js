import React from 'react';
import {shallow} from 'enzyme';

import Login from './Login';
import {Formik} from 'formik';
import {find} from 'async';

describe('Login', () => {
  let node;
  beforeEach(() => {
    node = shallow(<Login />);
  });

  it('should match formik snapshot', () => {
    expect(node).toMatchSnapshot();
  });

  it('should match form snapshot', () => {
    const form = node.instance().renderForm({
      isSubmitting: false,
      errors: {},
      handleChange: jest.fn(),
      handleSubmit: jest.fn()
    });

    expect(form).toMatchSnapshot();
  });

  it('should pass the form to Formik', () => {
    const renderMethod = node.instance().renderForm;
    expect(node.find(Formik).props().render).toEqual(renderMethod);
  });

  it('should render Formik', () => {
    expect(node.find(Formik).exists()).toBe(true);
  });

  it('should handle the form submit', () => {
    //
  });

  it('should pass the login data to api', () => {
    // call the login method.
  });

  it('should set initial values ', () => {
    //
  });
});
