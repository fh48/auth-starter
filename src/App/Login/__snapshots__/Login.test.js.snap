// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match form snapshot 1`] = `
<form
  className="form"
>
  <label
    className="form-field"
    htmlFor="email"
  >
    <span>
      E-mail:
    </span>
    <input
      name="email"
      onChange={[MockFunction]}
      type="email"
    />
  </label>
  <div
    className="form-field-error"
  />
  <label
    className="form-field"
    htmlFor="password"
  >
    <span>
      Password:
    </span>
    <input
      name="password"
      onChange={[MockFunction]}
      type="password"
    />
  </label>
  <div
    className="form-field-error"
  />
  <button
    onClick={[MockFunction]}
    type="submit"
    variant="contained"
  >
    Login
  </button>
</form>
`;

exports[`Login should match formik snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "enableReinitialize": false,
      "initialValues": null,
      "isInitialValid": false,
      "onSubmit": [Function],
      "render": [Function],
      "validateOnBlur": true,
      "validateOnChange": true,
      "validationSchema": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "enableReinitialize": false,
        "initialValues": null,
        "isInitialValid": false,
        "onSubmit": [Function],
        "render": [Function],
        "validateOnBlur": true,
        "validateOnChange": true,
        "validationSchema": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
